### PR TITLE
Finish germline table. Request to close Issue #6.

### PR DESCRIPTION
**Pull request by SenseCode**
1. Create GermlineTable.js;
2. Add changes for importing germlineTable in patient.js;
3. Add some  table features in _patientChart.scss.

### CODE REVIEW

File: src/encoded/schemas/germline.json [hunk 1]
```diff
@@ -0,0 +1,162 @@
+{
+    "title": "Germline Mutations",
+    "id": "/profiles/germline.json",
+    "description": "Schema for describing the germline mutation results.",
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "type": "object",
+    "required": ["patient", "register_date", "service_date", "target", "patient"],
```
Comment (vsmalladi) [inline src/encoded/schemas/germline.json]: Remove Second Patient reference in required

File: src/encoded/static/components/GermlineTable.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        this.tableConfig = {};
+        this.data = [];
+        this.germlineFilters = [];
+        this.transferData = [];
+    }
+
+
+    render() {
+        this.data = this.props.data;
+        //console.log(this.data);
```
Comment (vsmalladi) [inline src/encoded/static/components/GermlineTable.js]: @SenseCode  Remove this commment

**Review by vsmalladi — CHANGES_REQUESTED** [review]
Update code so that it is not requesting that data be updated.

Germline array will always be presnent in object.

File: src/encoded/static/components/GermlineTableNoState.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+import { SortTablePanel, SortTable } from './sorttable';
+
+class GermlineTableNoState extends React.Component {
+    constructor(props) {
+        super(props);
+        // this.state = {
+        //     showData: true,
+        // };
+        //this.tableConfig = {};
+        this.data = this.props.data;
```
Comment (cmonnom) [inline src/encoded/static/components/GermlineTableNoState.js]: Don't assign a property to a component variable. It's not a hard copy so you can't modify this.data anyway. Use this.props.data where this.data is used instead.

File: src/encoded/static/components/GermlineTableNoState.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                    <div className="chart-main">
+                        <div className="chart-title">
+                            <div><h3>{this.props.tableTitle}</h3> </div>
+                            <div><h4>Germline mutation data is not available!</h4></div>
+                        </div>
+                    </div>
+                </div>
+            );
+        } else {
+             this.germlineFilters = this.data.filter(i => (i.significance === 'Positive' || i.significance === 'Variant' || i.significance === 'Positive and Variant'));;
```
Comment (cmonnom) [inline src/encoded/static/components/GermlineTableNoState.js]: Should this line be removed?

File: src/encoded/static/components/GermlineTableNoState.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+            //this.transferData = this.germlineFilters.map(i => ({ values: [i.target, i.significance] }));
+            this.transferData = this.germlineFilters.map(i => ({ "target": i.target, "significance": i.significance}));
```
Comment (cmonnom) [inline src/encoded/static/components/GermlineTableNoState.js]: transferData doesn't seem to be the right variable name. Maybe you meant transformedData?

File: src/encoded/static/components/GermlineTableNoState.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+            console.log(this.data);
```
Comment (cmonnom) [inline src/encoded/static/components/GermlineTableNoState.js]: Remove console.log statements

File: src/encoded/static/components/GermlineTableNoState.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+            console.log(this.transferData);
+            //this.data = this.props.data;
+            
+            //console.log(this.germlineFilters);
+            // this.transferData=[{target: "VHL", significance: "Positive"}]
+            
+                if(this.transferData){
+                    const germlineTableColumns = {
```
Comment (cmonnom) [inline src/encoded/static/components/GermlineTableNoState.js]: Move const to the constructor

**Review by cmonnom — DISMISSED** [review]
Food for thought for the next code review:
Let's get rid of the existing GermlineTable.js and rename GermlineTableNoState.js into GermlineTable.js. Same with GermlineTableNoStatePanel.js -> GermlineTablePanel.js
Which one is the right file?
Remove code commented out and console.log statements

**Comment by vsmalladi** [comment]
> Food for thought for the next code review:
> Let's get rid of the existing GermlineTable.js and rename GermlineTableNoState.js into GermlineTable.js. Same with GermlineTableNoStatePanel.js -> GermlineTablePanel.js
> Which one is the right file?
> Remove code commented out and console.log statements

@cmonnom this code is still being resolved. Just pushing code before the weekend.

**Comment by cmonnom** [comment]
Maybe we can remove this branch from the pull request until it's ready so we don't get emailed everytime there is a commit?

**Comment by vsmalladi** [comment]
@cmonnom, unfortunately, a branch is linked directly to the merge request, so you can't uncouple it. Also, we want to track the review process and how we are addressing the changes requested. To deal with email's you can update your github settings, to only receive emails to a particular email for groups.

File: src/encoded/static/components/GermlineTableNoState.js (hunk 3 above)
Comment (SenseCode) [reply src/encoded/static/components/GermlineTableNoState.js]: I pulled a  wrong code and re-pushed on Saturday to correct it. Not sure how we can recall the wrong one back.  Sorry for the confusing. I will change the props. data and transformDataFunction . Any other suggestions are welcomed to address here.

Comment (vsmalladi) [reply src/encoded/static/components/GermlineTableNoState.js]: No need to recall. Its okay, this happens all the time.

File: src/encoded/static/components/germlineTable.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    filterData() {
+        this.germlineFilters = this.props.data.filter(i => (i.significance === 'Positive' || i.significance === 'Variant' || i.significance === 'Positive and Variant'));
+        this.transformData = this.germlineFilters.map(i => ({ target: i.target, significance: i.significance }));
+    }
+
+
+    renderData() {
+        if (!Array.isArray(this.props.data) || !this.props.data.length) {
+            return (<SortTablePanel title={this.props.tableTitle}>
+                        <div className="table-body"><h5>Germline mutation data is not available!</h5></div>
```
Comment (vsmalladi) [inline src/encoded/static/components/germlineTable.js]: Remove "!" form sentence.

File: src/encoded/static/components/germlineTable.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+                },
+            };
+            return (
+                <SortTablePanel title={this.props.tableTitle}>
+                    <SortTable list={this.transformData} columns={germlineTableColumns} />
+                </SortTablePanel>
+            );
+        }
+        return (<SortTablePanel title={this.props.tableTitle}>
+                    <div className="table-body"><h5>No Positive for mutations!</h5></div>
```
Comment (vsmalladi) [inline src/encoded/static/components/germlineTable.js]: Remove "!" form sentence.

File: src/encoded/static/components/patient.js [hunk 10]
```diff
@@ -35,18 +36,17 @@ class Patient extends React.Component {
             { id: 'Patients' },
             { id: <i>{context.accession}</i> },
         ];
-
+        console.log(context.germline);
```
Comment (vsmalladi) [inline src/encoded/static/components/patient.js]: Remove line

**Review by vsmalladi — APPROVED** [review]

**Pull request merged**